frontend/src/pages/hardwareProfiles/manage/validationUtils.ts

```diff
@@ -0,0 +1,155 @@
+
+  const checkNegative = (fieldName: string, value?: string | number) => {
+    if (value?.toString().startsWith('-')) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message: createIdentifierWarningMessage(
+          `${fieldName} for ${
+            identifier.resourceType ?? identifier.displayName
+          } cannot be negative.`,
+        ),
+        path: [fieldName],
+      });
+      return true;
+    }
+    return false;
+  };
+
+  const checkDecimal = (value: number, fieldName: string) => {
+    if (!Number.isInteger(value)) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message: createIdentifierWarningMessage(
+          `${fieldName} for ${
+            identifier.resourceType ?? identifier.displayName
+          } cannot be a decimal.`,
+        ),
+        params: { type: HardwareProfileWarningType.OTHER },
+      });
+    }
+  };
+
+  // 1. Validate minCount is within range
+  const minCountValidation = validateMinCount(identifier, unitOptions);
+  if (!minCountValidation.isValid) {
+    minCountValidation.issues?.forEach((issue) => ctx.addIssue(issue));
+  }
+
+  // 2. Validate defaultCount is within range
+  const defaultCountValidation = validateDefaultCount(identifier, unitOptions);
+  if (!defaultCountValidation.isValid) {
+    defaultCountValidation.issues?.forEach((issue) => ctx.addIssue(issue));
+  }
+
+  // 3. Validate maxCount is within range
+  const maxCountValidation = validateMaxCount(identifier, unitOptions);
+  if (!maxCountValidation.isValid) {
+    maxCountValidation.issues?.forEach((issue) => ctx.addIssue(issue));
+  }
+
+  // 3. Prevent negative values
+  const hasNegative = [
+    { value: identifier.minCount, name: 'Minimum count' },
+    { value: identifier.maxCount, name: 'Maximum count' },
+    { value: identifier.defaultCount, name: 'Default count' },
+  ].some(({ value, name }) => checkNegative(name, value));
+
+  if (hasNegative) {
+    return;
+  }
+
+  try {
+    const parseCount = (count: string | number) =>
+      splitValueUnit(count.toString(), unitOptions, true)[0];
+
+    const minCount = parseCount(identifier.minCount);
+    const maxCount = identifier.maxCount ? parseCount(identifier.maxCount) : undefined;
+    const defaultCount = parseCount(identifier.defaultCount);
+
+    // 4. Prevent decimal values
+    if (minCount !== undefined) {
+      checkDecimal(minCount, 'Minimum count');
+    }
+    if (defaultCount !== undefined) {
+      checkDecimal(defaultCount, 'Default count');
+    }
+    if (maxCount !== undefined) {
+      checkDecimal(maxCount, 'Maximum count');
+    }
+  } catch (e) {
+    ctx.addIssue({
+      code: z.ZodIssueCode.custom,
+      message: createIdentifierWarningMessage(
+        `The resource count for ${
+          identifier.resourceType ?? identifier.displayName
+        } has an invalid unit.`,
+      ),
+      params: { type: HardwareProfileWarningType.OTHER },
```

Comment:
instead of params can we put the custom type in code... we probably dont need an other type anymore and just use the missing cpu type as thats the only specific one